pkg/prometheus/statefulset.go

```diff
@@ -440,7 +439,18 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMaps []
 			Port: intstr.FromString("web"),
 		},
 	}
-
+	podAnnotations := map[string]string{}
+	podLabels := map[string]string{}
+	for k, v := range p.Spec.PodMetadata.Labels {
+		// ToDo: define which labels can be set and sanitize them.
```

Comment:
I think we can skip this sanitizing step, but make clear, that if these annotations/labels are used to enable alpha features, they are unsupported clusters and there is no guarantee that it will work.